servant-auth-server/src/Servant/Auth/Server/Internal/ConfigTypes.hs

```diff
@@ -79,7 +75,31 @@ defaultCookieSettings = CookieSettings
     , cookiePath        = Just "/"
     , cookieSameSite    = SameSiteLax
     , sessionCookieName = "JWT-Cookie"
-    , xsrfCookieName    = "XSRF-TOKEN"
+    , xsrfSetting       = XsrfCookie def
+    }
+
+data XsrfSetting = XsrfCookie XsrfCookieSettings
```

Comment:
Can you add some documentation to this datatype and it's constructors? I'd link to the [OWASP page](https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)_Prevention_Cheat_Sheet).